runtime/src/accounts_db.rs

```diff
@@ -573,12 +576,59 @@ impl AccountsDB {
         let mut reclaims = Vec::new();
         let mut accounts_index = self.accounts_index.write().unwrap();
         for purge in &purges {
-            reclaims.extend(accounts_index.purge(purge));
+            reclaims.extend(accounts_index.would_purge(purge));
         }
         let last_root = accounts_index.last_root;
+
+        // Calculate store counts as if everything was purged
+        // Then purge if we can
+        let mut store_counts: HashMap<AppendVecId, usize> = HashMap::new();
+        let storage = self.storage.read().unwrap();
+        for (slot_id, account_info) in &reclaims {
+            if let Some(slot_storage) = storage.0.get(&slot_id) {
+                if let Some(store) = slot_storage.get(&account_info.id) {
+                    if let Some(store_count) = store_counts.get_mut(&account_info.id) {
+                        *store_count -= 1;
+                    } else {
+                        store_counts
+                            .insert(account_info.id, store.count_and_status.read().unwrap().0);
```

Comment:
Should this start with `.insert(...count... **- 1**);`, shouldn't?

Locally, both version passes the relevant test, indicating still not adequate test coverage. ;)

Reply:
nits: use `store.id` to make reasoning a bit lighter? (first of all, `id` is too generic...)

Reply:
Thanks for the catch. I added a test and changed AccountIndex::id to AccountIndex::store_id.